blocks/embed/embed.js

```diff
@@ -0,0 +1,222 @@
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+/* global window */
+/* eslint-disable import/named, import/extensions */
+
+import { buildFigure } from '../../scripts/scripts.js';
+
+const loadScript = (url, callback, type) => {
+    const $head = document.querySelector('head');
+    const $script = document.createElement('script');
+    $script.src = url;
+    if (type) {
+      $script.setAttribute('type', type);
+    }
+    $head.append($script);
+    $script.onload = callback;
+    return $script;
+}
+
+// 'open.spotify.com' returns 'spotify'
+const getServer = (url) => {
+    const l = url.hostname.lastIndexOf('.');
+    return url.hostname.substring(url.hostname.lastIndexOf('.', l - 1) + 1, l);
+}
+  
+const getDefaultEmbed = (url) => {
+    return `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;">
+      <iframe src="${url.href}" style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" allowfullscreen=""
+        scrolling="no" allow="encrypted-media" title="Content from ${url.hostname}" loading="lazy">
+      </iframe>
+    </div>`;
+}
+  
+const embedYoutube = (url) => {
+    const usp = new URLSearchParams(url.search);
+    const vid = usp.get('v');
+    const embed = url.pathname;
+    const embedHTML = `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;">
+      <iframe src="https://www.youtube.com${vid ? `/embed/${vid}?rel=0&amp;v=${vid}` : embed}" style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" allow="encrypted-media; accelerometer; gyroscope; picture-in-picture" allowfullscreen="" scrolling="no" title="Content from Youtube" loading="lazy"></iframe>
+    </div>`;
+    return embedHTML;
+}
+  
+const embedInstagram = (url) => {
+    const location = window.location.href;
+    const src = `${url.origin}${url.pathname}${url.pathname.charAt(url.pathname.length - 1) === '/' ? '' : '/'}embed/?cr=1&amp;v=13&amp;wp=1316&amp;rd=${location.endsWith('.html') ? location : `${location}.html`}`;
+    const embedHTML = `<div style="width: 100%; position: relative; display: flex; justify-content: center">
+      <iframe class="instagram-media instagram-media-rendered" id="instagram-embed-0" src="${src}"
+        allowtransparency="true" allowfullscreen="true" frameborder="0" height="530" style="background: white; border-radius: 3px; border: 1px solid rgb(219, 219, 219);
+        box-shadow: none; display: block;" loading="lazy">
+      </iframe>
+    </div>`;
+    return embedHTML;
+}
+  
+const embedVimeo = (url) => {
+    const linkArr = url.href.split('/');
```

Comment:
i am not sure i understand this. it seems that in the case where `linkArr` is `falsy` it would arguably assign `video` with undefined and otherwise video would be assigned with the second path segment? 

the code seems to be somewhat equivalent to:

```
const video = url.pathname.split('/')[2]
```
also lacking the guards for the pathname not having enough segments?
and still lacks the guard for the